src/app/page.tsx

```diff
@@ -63,9 +63,9 @@ export default async function Main({ searchParams }: { searchParams: { token: st
               'use server'
               await handleCreate(token, title, description, workflowStateId, assigneeId, assigneeType)
             }}
-            updateWorkflowStateIdOfTask={async (taskId, targetWorkflowStateId) => {
+            updateTask={async (taskId: string, payload: UpdateTaskRequest) => {
               'use server'
-              await updateWorkflowStateIdOfTask(token, taskId, targetWorkflowStateId)
+              await updateTask({ token, taskId, payload })
```

Comment:
let's organize actions related to specific pages in separate files instead of using them inline like this.

This provides a better separation of concerns between UI logic and actions

Reply:
Action is not being used inline here. `updateTask` action is already in a separate file. If this was what you meant to say by inline here. Otherwise, please correct me. 

For instance, `updateTask` action should be used from both `/page.tsx` to update the assignee and `/detail/page.tsx`, to update other parameters. For this reason, organizing actions related to specific pages in separate files may again repeat the same actions. 

I found that requests for same endpoint are being repeated if you observe in the `actions.ts` files, which is not necessary. So I was thinking about a new approach in organizing the actions. I think it  is better to organize the actions according to the API requests. For instance, for all the tasks related CRUD operations, keeping actions for POST, PATCH, and DELETE in a single `task-action.ts` file inside the `/action` folder.

Reply:
If updateTask is already a server action, is there a reason why we are repeating 'use server' again? I'm assuming you have 'use server' at the top of your actions file already.

Reply:
It seems like 'use server' directive is necessary here otherwise I am getting error as
```
Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server". Or maybe you meant to call this function rather than return it.
  <... handleCreate={function handleCreate} updateTask=...>
```

Reply:
You could try directly passing it into the funciton without the async function wrapper using three separate arguments instead of a single object